apis/manifests/v1alpha1/packagemanifestlock_types.go

```diff
@@ -0,0 +1,71 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/strings/slices"
+)
+
+// +kubebuilder:object:root=true
+type PackageManifestLock struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PackageManifestLockSpec `json:"spec,omitempty"`
+}
+
+type PackageManifestLockSpec struct {
+	// List of resolved images
+	Images []PackageManifestLockImage `json:"images"`
+}
+
+// PackageManifestLockImage contains information about a resolved image
+type PackageManifestLockImage struct {
+	// Image name to be use to reference it in the templates
+	Name string `json:"name"`
+	// Image identifier (REPOSITORY[:TAG])
+	Image string `json:"image"`
+	// Image digest
+	Digest string `json:"digest"`
+}
+
+// Validates the PackageManifestLock and returns an aggregated list of all errors.
+func (m *PackageManifestLock) Validate() error {
+
+	var allErrs field.ErrorList
+
+	specImages := field.NewPath("spec").Child("images")
+	if len(m.Spec.Images) > 0 {
```

Comment:
See above regarding removing the condition and the switch

Reply:
Thx for the suggestion! Changed :+1: